File: Iniciante/1047 - Tempo de Jogo com Minutos/Python/1047.py
TOTAL_MINUTES_IN_A_HOUR = 60
TOTAL_MINUTES_IN_A_DAY = 1440


def main():
    start_hour, start_minute, final_hour, final_minute = map(int, input().split())

    hours, minutes = calculate_game_duration(
        start_hour, start_minute, final_hour, final_minute
    )

    print(f"O JOGO DUROU {hours} HORA(S) E {minutes} MINUTO(S)")


def calculate_game_duration(start_hour, start_minute, final_hour, final_minute):
    start_total_minutes = start_hour * TOTAL_MINUTES_IN_A_HOUR + start_minute
    end_total_minutes = final_hour * TOTAL_MINUTES_IN_A_HOUR + final_minute

    duration_in_minutes = end_total_minutes - start_total_minutes

    if duration_in_minutes <= 0:
        duration_in_minutes += TOTAL_MINUTES_IN_A_DAY

    return convert_time(duration_in_minutes)


def convert_time(time):
    return divmod(time, TOTAL_MINUTES_IN_A_HOUR)


if __name__ == "__main__":
    main()

File: Iniciante/1047 - Tempo de Jogo com Minutos/TypeScript/1047.ts
import * as fs from "fs";

const TOTAL_MINUTES_IN_A_HOUR: number = 60;
const TOTAL_MINUTES_IN_A_DAY: number = 1440;

function main(): void {
    const input: string = fs.readFileSync("/dev/stdin", "utf8");
    const [startHour, startMinute, finalHour, finalMinute] = input.trim().split(/\s+/).map(Number);
    const [hours, minutes] = calculateGameDuration(startHour,startMinute,finalHour,finalMinute); 
    console.log(`O JOGO DUROU ${hours} HORA(S) E ${minutes} MINUTO(S)`);
}

function calculateGameDuration(startHour: number, startMinute: number, finalHour: number, finalMinute: number): [number, number] {
    const startTotalMinutes: number = startHour * TOTAL_MINUTES_IN_A_HOUR + startMinute;
    const endTotalMinutes: number = finalHour * TOTAL_MINUTES_IN_A_HOUR + finalMinute;

    let durationInMinutes: number = endTotalMinutes - startTotalMinutes;

    if (durationInMinutes <= 0) {
        durationInMinutes += TOTAL_MINUTES_IN_A_DAY;
    }

    return convertTime(durationInMinutes);
}

function convertTime(time: number): [number, number] {
    const hour: number = Math.floor(time / TOTAL_MINUTES_IN_A_HOUR);
    const minutes: number = time % TOTAL_MINUTES_IN_A_HOUR;
    return [hour, minutes];
}

main();